### PR TITLE
qa/deepsea: add cram test to tier3/rbd

### PR DESCRIPTION
**Pull request by smithfarm**
Signed-off-by: Nathan Cutler <ncutler@suse.com>

### CODE REVIEW

**Comment by smithfarm** [comment]
Testing with command: `SUITE="deepsea/tier3/rbd"; SUITE_REPO="https://github.com/SUSE/ceph.git"; SUITE_BRANCH="wip-qa-cram"; FILTER="cram.yaml"; FILTER_OUT="opensuse"; NAME="$USER"; teuthology-openstack --verbose --ceph-repo $SUITE_REPO --ceph $SUITE_BRANCH --suite-repo $SUITE_REPO --suite-branch $SUITE_BRANCH --filter $FILTER --filter-out $FILTER_OUT --key-filename /home/$USER/.ssh/storage-automation --key-name storage-automation --name $NAME --suite $SUITE --test-repo 'internal!2:http://10.86.0.120/artifacts/ci/a3d46e1a52d4991a98320b58f94c803c205426cd7bd8e22d517f9e63f6632150/SUSE-Enterprise-Storage-6-POOL-Internal-x86_64-Build65.3' --test-repo 'ses!2:http://10.86.0.120/artifacts/ci/3f2f40649774c5dff663e7871276b33b5f0e4a60e10cafb0f53dada88c38cac8/SUSE-Enterprise-Storage-6-POOL-x86_64-Build65.4' --test-repo basesystem:http://10.86.0.120/artifacts/ci/cc6b0df108e4543f7eac96c9afbbf00f53959cdef7d5726626af15685a30b377/SLE-15-SP1-Module-Basesystem-POOL-x86_64-Build228.2 --test-repo server:http://10.86.0.120/artifacts/ci/a3607fcccdb2b23107df666bfb2c8d85665a948f1da049d9021bd037c020ba02/SLE-15-SP1-Module-Server-Applications-POOL-x86_64-Build228.2`

NOTE: The test failed because we don't have `python3-cram` on the Internal media. As soon as we have it, I'll update the command and re-run it.

**Comment by smithfarm** [comment]
@shyukri This should now work - i.e. as soon as you add a file `src/test/cli-integration/rbd/diff.t` this test will run it.

File: src/test/cli-integration/rbd/diff.t [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  []
+  $ rbd diff --from-snap=allzeroes xrbddiff1/xtestdiff1 --format json
+  []
+  $ rbd bench --io-type write --io-size 1M --io-total 1M xrbddiff1/xtestdiff1 > /dev/null 2>&1
+  $ rbd diff xrbddiff1/xtestdiff1 --format json
+  [{"offset":0,"length":1048576,"exists":"true"}]
+  $ rbd diff --from-snap=allzeroes xrbddiff1/xtestdiff1 --format json
+  [{"offset":0,"length":1048576,"exists":"true"}]
+  $ rbd snap create xrbddiff1/xtestdiff1 --snap=snap1
+  $ rbd snap list xrbddiff1/xtestdiff1 --format json | python -mjson.tool | sed 's/,$/, /'
```
Comment (smithfarm) [inline src/test/cli-integration/rbd/diff.t]: change `python` to `python3` here

**Comment by smithfarm** [comment]
@shyukri Please change the title of your commit to:

```
qa/deepsea: add diff.t to rbd cram test
```

**Comment by shyukri** [comment]
Results: http://10.86.3.20/ubuntu-2019-05-30_14:30:52-deepsea:tier3:rbd-wip-qa-cram---basic-openstack/3/teuthology.log

**Comment by smithfarm** [comment]
Huh, cram only produces verbose output when something goes wrong. When the test passes, it is very terse :-)

```
2019-05-30T14:52:20.659 INFO:tasks.cram.client.0.target192168000040.stdout:/home/ubuntu/cephtest/archive/cram.client.0/diff.t: passed
2019-05-30T14:52:20.660 INFO:tasks.cram.client.0.target192168000040.stdout:# Ran 1 tests, 0 skipped, 0 failed.
```

**Comment by smithfarm** [comment]
This is fine, but I'm going to hold off merging it until after M17.

**Pull request merged**